### ISSUE TITLE
Without CasA or multi-stage architecture,  loss from SESSD and data augmentation from SESSD, your propose really work well?

### ISSUE THREAD

**Issue by 3xCode** [hailanyi/TED#2]
To my extensive experiments, what really works is the CasA multi-stage module and the data augmentation from SESSD which you did not mention in your paper, did you condut ablation experiments on SECOND or other single-stage framework with your proposed modules? I do not think they will work well without CasA multi-stage module and the data augmentation from SESSD.

It is very easy to achieve a good experiment results on KITTI test set. You just add CasA and data augmentation from SESSD.Then, add something else...you know... The results will be very nice, then you can write a good paper ^_^

I have doubt that whether you used the loss from SESSD, it useful to improve the moderate and hard performance, but it does not exist in your source code. It will lead to not be reproduced ^_^

**Comment by hailanyi** [comment]
Thanks for your question, I did not conducted an experiment on SECOND or others. From my past experiments, the CasA and augmentation from SESSD are optional.  However, the hyper-parameters of the released codes are highly optimised with the two parts (I hope our CasA and TED can work together for better and more stable performance). The released TED-M can attain 89+ AP during every training. The version of our paper can only attain 88+ in most training.

Important notes:
If you directly remove some parts, the detection recall well drop dramatically, the performance is also decreased ~2% AP. So if you remove or replace some parts, you need to adjust some hyper-parameters of DA-Aug and TEDhead (score threshold, proposal number and etc.) to ensure a good recall and AP. 

'The loss from SESSD' means the ODiou loss? I did not use this loss, because it works every slow (need 3X time to train the network). Instead, I used a fake IoU loss (bb_loss in pcdet/utils/bbloss.py).  It can improve the recall and work very fast, but it improve the AP marginally, So I did not mentioned in paper.

Again, thanks for your question. In deed, the CasA+TED can attain better results on KITTI test set. We already developed a better detector based on this framework (~91 AP on KITTI val set, paper is preparing). I hope others can also investigate the potentials, so we chose to open source this code.
